cityofaustin/atd-moped
# 6233: Remove unused tables and rework FK relationships to allow cascading deletions of child table records

## Associated issues

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/6233.

## Considerations

This PR changes the way that the database will handle a `DELETE` command on `moped_project` and its child tables. It configures the database to `CASCADE` the delete command down to get all related data of a project when a project is deleted. 

This PR allows for a project and all related data to be deleted with a single SQL command or mutation.

It also includes `DROP`ing a number of unused tables.

## Testing

This PR is entirely related to database changes, so testing must occur locally or on the _testing_ deployment. 

For those who do run moped locally, I recommend doing a round of local testing and then doing a round of testing on the _test_ instance. Testing on _test_ may require use of the bastion host to access the DB instance as it is firewalled behind it.

A portion of the testing will include issuing a `DELETE` command against the `moped_project` table, and I think we need to be mindful of preserving that data on the shared environments until it needs to be deleted. Let's keep communicating about database changes that remove or alter data outside of our local instances.

All that said, the testing URL is: https://moped-test.austinmobility.io/moped/

**Steps to test:**

* Ensure migrations run when you spin up an instance of the database. This tests the changes to the seed data.
* Ensure that just the new migrations run against a restoration of data into your local database stack from both staging and production.
* Ensure that the correct tables have been removed.
* Look at the tables that have the foreign key constraints changed to use cascading deletes. Make sure there are no tables that contain information that is part of the app (like a list of departments, divisions, etc.) These deletes should only cascade into tables which contain information about a project or related, user-entered, data.
* Make sure that you can `DELETE` any `moped_project` that you want or can construct in a single SQL statement without errors.

---
#### Ship list
